packages/core/src/types.ts

```diff
@@ -678,6 +678,15 @@ export interface MachineConfig<
    * The machine's own version.
    */
   version?: string;
+  schema?: MachineSchema<TContext, TEvent>;
+}
+
+export interface MachineSchema<TContext, TEvent extends EventObject> {
+  context?: TContext;
+  events?: TEvent;
+  actions?: ActionObject<TContext, TEvent>;
```

Comment:
I'm not sure this is a proper type for this - this should focus on the schema, so on a serializable `{ type: string, [key: string]: any }`. And this type includes an `exec` property and stuff.

I imagine that in the future we plan to put a schema for additional action parameters here.